### PR TITLE
tools: update tools faq

### PR DESCRIPTION
**Pull request by glorv**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Add a faq for reparo

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by glorv** [comment]
@3pointer @july2993 @overvenus @kennytm   PTAL

**Review by Little-Wallace — APPROVED** [review]
LGTM

File: tidb-binlog/tidb-binlog-faq.md [hunk 1]
```diff
@@ -216,3 +216,18 @@ Pump 以 `paused` 状态退出进程时，不保证所有 binlog 数据被下游
 ## 可以使用 `change pump`、`change drainer` 等 SQL 操作来暂停或者下线 Pump/Drainer 服务吗？
 
 目前还不支持。这种 SQL 操作会直接修改 PD 中保存的状态，在功能上等同于使用 binlogctl 的 `update-pump`、`update-drainer` 命令。如果需要暂停或者下线，仍然要使用 binlogctl。
+
+## TiDB 写 binlog 失败导致 TiDB 卡住，日志不停报错 "listener stopped, waiting for manual stop"
```
Comment (overvenus) [inline tidb-binlog/tidb-binlog-faq.md:220]: PTAL @july2993, thanks!

Comment (july2993) [reply tidb-binlog/tidb-binlog-faq.md:220]: 现在是什么行为

File: tidb-binlog/tidb-binlog-faq.md [hunk 2]
```diff
@@ -216,3 +216,18 @@ Pump 以 `paused` 状态退出进程时，不保证所有 binlog 数据被下游
...
+
+在 3.0.12 之前版本的 tidb，写 binlog 失败会导致 tidb 报 fatal error， 但是 tidb 不会自动退出只是停止服务，看起来像是服务卡住, tidb 的日志里面可以看到 “listener stopped, waiting for manual stop”。 遇到此问题需要根据具体情况判断具体是什么原因导致写 binlog 失败，如果是下游 binlog 写入缓慢导致，可以考虑扩容 pump 或增加写 binlog 超时时间。
```
Comment (july2993) [inline tidb-binlog/tidb-binlog-faq.md:222]: 改了的话建议说下新版本的行为

**Review by july2993 — COMMENTED** [review]
rest LGTM

**Review by kennytm — COMMENTED** [review]
LGTM (for lightning part)

File: tidb-binlog/tidb-binlog-faq.md (hunk 2 above)
Comment (glorv) [reply tidb-binlog/tidb-binlog-faq.md:222]: updated. PTAL again @july2993

File: tidb-binlog/tidb-binlog-faq.md [hunk 3]
```diff
@@ -221,6 +221,8 @@ Pump 以 `paused` 状态退出进程时，不保证所有 binlog 数据被下游
 
 在 3.0.12 之前版本的 tidb，写 binlog 失败会导致 tidb 报 fatal error， 但是 tidb 不会自动退出只是停止服务，看起来像是服务卡住, tidb 的日志里面可以看到 “listener stopped, waiting for manual stop”。 遇到此问题需要根据具体情况判断具体是什么原因导致写 binlog 失败，如果是下游 binlog 写入缓慢导致，可以考虑扩容 pump 或增加写 binlog 超时时间。
 
+新版本的 tidb 已优化此逻辑，写入 binlog 失败将使事务执行失败返回而不是导致 tidb 直接报错退出。
```
Comment (july2993) [inline tidb-binlog/tidb-binlog-faq.md:224]: 按前面讲的是停止服务 不退出

Comment (glorv) [reply tidb-binlog/tidb-binlog-faq.md]: fixed

Comment (glorv) [reply tidb-binlog/tidb-binlog-faq.md]: @july2993  PTAL again, thanks~

**Review by july2993 — APPROVED** [review]

File: tidb-binlog/tidb-binlog-faq.md [hunk 4]
```diff
@@ -216,3 +216,20 @@ Pump 以 `paused` 状态退出进程时，不保证所有 binlog 数据被下游
...
+
+## TiDB 写 binlog 失败导致 TiDB 卡住，日志不停报错 "listener stopped, waiting for manual stop"
+
+在 3.0.12 之前版本的 tidb，写 binlog 失败会导致 tidb 报 fatal error， 但是 tidb 不会自动退出只是停止服务，看起来像是服务卡住, tidb 的日志里面可以看到 “listener stopped, waiting for manual stop”。 遇到此问题需要根据具体情况判断具体是什么原因导致写 binlog 失败，如果是下游 binlog 写入缓慢导致，可以考虑扩容 pump 或增加写 binlog 超时时间。
+
+新版本的 tidb 已优化此逻辑，写入 binlog 失败将使事务执行失败返回而不会导致 tidb 卡住。
+
+## TiDB 向 pump 写入了重复的 binlog？
+
+TiDB 在写入 binlog 失败或者超时的情况下，会重试下一个可用的 pump 直到写入成功。所以如果某个 pump 写入较慢， 导致 tidb 超时(默认 15s), 此时 tidb 判定写入失败并尝试下一个 pump。 如果超时的 pump 实际也写入成功，则会出现同一条 binlog 被写入到多个 pump。Drainer 在处理 binlog 的时候，会自动去重 tso 相同的 binlog，所以这种重复的写入，对下游无感知，不会对同步逻辑产生影响。
```
Comment (TomShawn) [inline tidb-binlog/tidb-binlog-faq.md:228]: 请规范专有名称的大小写以及完整写法：
+ tidb --> TiDB
+ pump --> Pump
+ Lightning --> TiDB Lightning

File: tidb-binlog/tidb-binlog-faq.md [hunk 5]
```diff
@@ -216,3 +216,20 @@ Pump 以 `paused` 状态退出进程时，不保证所有 binlog 数据被下游
...
+
+## 在使用全量+增量方式恢复的过程中，reparo 中断了，可以使用日志里面最后一个 tso 接起来吗？
+
+可以。reparo 不会在启动时自动开启 safe-mode 模式，需要手动操作：
+1、Reparo 中断后，记录日志中最后一个 tso，记为 checkpoint tso。
```
Comment (TomShawn) [inline tidb-binlog/tidb-binlog-faq.md:233]: 请将顿号`、`替换为 `. `
```suggestion
1. Reparo 中断后，记录日志中最后一个 tso，记为 checkpoint tso。
```

File: tidb-binlog/tidb-binlog-faq.md [hunk 6]
```diff
@@ -216,3 +216,20 @@ Pump 以 `paused` 状态退出进程时，不保证所有 binlog 数据被下游
...
+2、修改 Reparo 配置文件，将 start-tso 设为 checkpoint tso + 1，将 stop-tso 设为 checkpoint tso + 80,000,000,000（大概是 checkpoint tso 延后 5 分钟），将 safe-mode 设为 true，启动 reparo，reparo 会将数据同步到 stop-tso 后自动停止。
```
Comment (TomShawn) [inline tidb-binlog/tidb-binlog-faq.md:234]: 变量/参数/配置名以及值，请用行内代码反引号包裹。

File: tidb-lightning/tidb-lightning-faq.md [hunk 7]
```diff
@@ -340,3 +340,18 @@ tidb-lightning-ctl --config conf/tidb-lightning.toml --checkpoint-error-destroy=
 
 - 如果集群中有 TiFlash 节点，可以将集群更新到 4.0.0-rc.2 或更新版本。
 - 如果不方便升级，可以临时禁用 TiFlash。
+
+## `tidb lightning encountered error: TiDB version too old, expected '>=4.0.0', found '3.0.18'`
+
+Lightning Local Backend 只支持导入到 4.0.0 以上的 TiDB 集群，如果试图使用 Local Backend 导入 2.x 或 3.x 的集群，就会报如上错误。遇到这个情况，请修改配置使用 Importer 或 TiDB Backend。
```
Comment (TomShawn) [inline tidb-lightning/tidb-lightning-faq.md:346]: 根据 https://docs.pingcap.com/zh/tidb/dev/tidb-lightning-backends 的写法，请规范写为 Local-backend，Importer-backend 和 TiDB-backend。

File: tidb-binlog/tidb-binlog-faq.md [hunk 8]
```diff
@@ -216,3 +216,21 @@ Pump 以 `paused` 状态退出进程时，不保证所有 binlog 数据被下游
 ## 可以使用 `change pump`、`change drainer` 等 SQL 操作来暂停或者下线 Pump/Drainer 服务吗？
 
 目前还不支持。这种 SQL 操作会直接修改 PD 中保存的状态，在功能上等同于使用 binlogctl 的 `update-pump`、`update-drainer` 命令。如果需要暂停或者下线，仍然要使用 binlogctl。
+
+## TiDB 写 binlog 失败导致 TiDB 卡住，日志不停报错 "listener stopped, waiting for manual stop"
+
+在 3.0.12 之前版本的 tidb，写 binlog 失败会导致 tidb 报 fatal error， 但是 tidb 不会自动退出只是停止服务，看起来像是服务卡住, tidb 的日志里面可以看到 “listener stopped, waiting for manual stop”。 遇到此问题需要根据具体情况判断具体是什么原因导致写 binlog 失败，如果是下游 binlog 写入缓慢导致，可以考虑扩容 pump 或增加写 binlog 超时时间。
```
Comment (TomShawn) [inline tidb-binlog/tidb-binlog-faq.md:222]:
```suggestion
在 TiDB 3.0.12 之前，binlog 写入失败会导致 TiDB 报 fatal error。但是 TiDB 不会自动退出只是停止服务。看起来像服务卡住。TiDB 日志中可看到 "listener stopped, waiting for manual stop"。 遇到该问题需要根据具体情况判断是什么原因导致 binlog 写入失败。如果是下游 binlog 写入缓慢导致的，可以考虑扩容 Pump 或增加写 binlog 的超时时间。
```

File: tidb-binlog/tidb-binlog-faq.md [hunk 9]
```diff
@@ -216,3 +216,21 @@ Pump 以 `paused` 状态退出进程时，不保证所有 binlog 数据被下游
 ## 可以使用 `change pump`、`change drainer` 等 SQL 操作来暂停或者下线 Pump/Drainer 服务吗？
 
 目前还不支持。这种 SQL 操作会直接修改 PD 中保存的状态，在功能上等同于使用 binlogctl 的 `update-pump`、`update-drainer` 命令。如果需要暂停或者下线，仍然要使用 binlogctl。
+
+## TiDB 写 binlog 失败导致 TiDB 卡住，日志不停报错 "listener stopped, waiting for manual stop"
```
Comment (TomShawn) [inline tidb-binlog/tidb-binlog-faq.md:220]:
```suggestion
## TiDB 写入 binlog 失败导致 TiDB 卡住，日志中出现 `listener stopped, waiting for manual stop` 错误
```

File: tidb-binlog/tidb-binlog-faq.md [hunk 10]
```diff
@@ -216,3 +216,21 @@ Pump 以 `paused` 状态退出进程时，不保证所有 binlog 数据被下游
...
+
+在 3.0.12 之前版本的 tidb，写 binlog 失败会导致 tidb 报 fatal error， 但是 tidb 不会自动退出只是停止服务，看起来像是服务卡住, tidb 的日志里面可以看到 “listener stopped, waiting for manual stop”。 遇到此问题需要根据具体情况判断具体是什么原因导致写 binlog 失败，如果是下游 binlog 写入缓慢导致，可以考虑扩容 pump 或增加写 binlog 超时时间。
+
+新版本的 tidb 已优化此逻辑，写入 binlog 失败将使事务执行失败返回而不会导致 tidb 卡住。
```
Comment (TomShawn) [inline tidb-binlog/tidb-binlog-faq.md:224]: + 新版本是 3.0.12 吗？
+ tidb --> TiDB

File: tidb-binlog/tidb-binlog-faq.md [hunk 11]
```diff
@@ -216,3 +216,21 @@ Pump 以 `paused` 状态退出进程时，不保证所有 binlog 数据被下游
...
+
+## TiDB 向 pump 写入了重复的 binlog？
+
+TiDB 在写入 binlog 失败或者超时的情况下，会重试下一个可用的 Pump 直到写入成功。所以如果某个 Pump 写入较慢， 导致 TiDB 超时(默认 15s), 此时 TiDB 判定写入失败并尝试下一个 Pump。如果超时的 Pump 实际也写入成功，则会出现同一条 binlog 被写入到多个 Pump。Drainer 在处理 binlog 的时候，会自动去重 tso 相同的 binlog，所以这种重复的写入，对下游无感知，不会对同步逻辑产生影响。
```
Comment (TomShawn) [inline tidb-binlog/tidb-binlog-faq.md:228]:
```suggestion
TiDB 在写入 binlog 失败或者超时的情况下，会重试将 binlog 写入到下一个可用的 Pump 节点直到写入成功。所以如果写入到某个 Pump 节点较慢，导致 TiDB 超时（默认 15s），此时 TiDB 判定写入失败并尝试下一个 Pump 节点。如果超时的 Pump 实际也写入成功，则会出现同一条 binlog 被写入到多个 Pump 节点。Drainer 在处理 binlog 的时候，会自动去重 TSO 相同的 binlog，所以这种重复的写入，对下游无感知，不会对同步逻辑产生影响。
```

File: tidb-binlog/tidb-binlog-faq.md [hunk 12]
```diff
@@ -216,3 +216,21 @@ Pump 以 `paused` 状态退出进程时，不保证所有 binlog 数据被下游
...
+
+## 在使用全量+增量方式恢复的过程中，reparo 中断了，可以使用日志里面最后一个 tso 接起来吗？
```
Comment (TomShawn) [inline tidb-binlog/tidb-binlog-faq.md:230]: + 把 Reparo 接起来吗？
+ 大小写问题，下文中相同

File: tidb-lightning/tidb-lightning-faq.md [hunk 13]
```diff
@@ -340,3 +340,18 @@ tidb-lightning-ctl --config conf/tidb-lightning.toml --checkpoint-error-destroy=
 
 - 如果集群中有 TiFlash 节点，可以将集群更新到 4.0.0-rc.2 或更新版本。
 - 如果不方便升级，可以临时禁用 TiFlash。
+
+## `tidb lightning encountered error: TiDB version too old, expected '>=4.0.0', found '3.0.18'`
+
+TiDB Lightning Local-backend 只支持导入到 `4.0.0` 以上的 TiDB 集群，如果试图使用 Local-backend 导入 `2.x` 或 `3.x` 的集群，就会报如上错误。遇到这个情况，请修改配置使用 Importer-backend 或 TiDB-backend。
```
Comment (TomShawn) [inline tidb-lightning/tidb-lightning-faq.md:346]:
```suggestion
TiDB Lightning Local-backend 只支持导入到 v4.0.0 以上的 TiDB 集群。如果尝试使用 Local-backend 导入 v2.x 或 v3.x 的集群，就会报以上错误。此时可以修改配置使用 Importer-backend 或 TiDB-backend。
```

File: tidb-lightning/tidb-lightning-faq.md [hunk 14]
```diff
@@ -340,3 +340,18 @@ tidb-lightning-ctl --config conf/tidb-lightning.toml --checkpoint-error-destroy=
...
+
+部分 `nightly` 版本的 TiDB 集群的版本可能类似 "4.0.0-beta.2" 这样，这样的版本实际支持 Local-backend，遇到这个情况，可以通过设置配置  `check-requirements = false` 跳过版本检查。在设置此参数之前，请务必确保 TiDB Lightning 的配置是支持对应的版本，否则无法保证导入成功。
```
Comment (TomShawn) [inline tidb-lightning/tidb-lightning-faq.md:348]: 看不太明白 ``` 部分 `nightly` 版本的 TiDB 集群的版本可能类似 "4.0.0-beta.2" 这样 ```
```suggestion
部分 `nightly` 版本的 TiDB 集群的版本可能类似 "4.0.0-beta.2" 这样，这样的版本实际支持 Local-backend。如果使用 `nightly` 版本遇到该报错，可以通过设置配置  `check-requirements = false` 跳过版本检查。在设置此参数之前，请确保 TiDB Lightning 的配置支持对应的版本，否则无法保证导入成功。
```

File: tidb-lightning/tidb-lightning-faq.md [hunk 15]
```diff
@@ -340,3 +340,18 @@ tidb-lightning-ctl --config conf/tidb-lightning.toml --checkpoint-error-destroy=
...
+
+## `restore table test.district failed: unknown columns in header [...]`
+
+通常出现这种错误是因为 CSV 格式的数据文件不包含 header (第一行也是数据)，因此需要在 Lightning 的配置文件中增加如下配置项：
```
Comment (TomShawn) [inline tidb-lightning/tidb-lightning-faq.md:352]:
```suggestion
出现该错误通常是因为 CSV 格式的数据文件不包含 header (第一行也是数据)，因此需要在 TiDB Lightning 的配置文件中增加如下配置项：
```

File: tidb-lightning/tidb-lightning-faq.md [hunk 16]
```diff
@@ -340,3 +340,18 @@ tidb-lightning-ctl --config conf/tidb-lightning.toml --checkpoint-error-destroy=
 
 - 如果集群中有 TiFlash 节点，可以将集群更新到 4.0.0-rc.2 或更新版本。
 - 如果不方便升级，可以临时禁用 TiFlash。
+
+## `tidb lightning encountered error: TiDB version too old, expected '>=4.0.0', found '3.0.18'`
+
+TiDB Lightning Local-backend 只支持导入到 v4.0.0 以上的 TiDB 集群，如果尝试使用 Local-backend 导入 v2.x 或 v3.x 的集群，就会报以上错误。此时可以修改配置使用 Importer-backend 或 TiDB-backend 进行导入。
```
Comment (TomShawn) [inline tidb-lightning/tidb-lightning-faq.md:346]:
```suggestion
TiDB Lightning Local-backend 只支持导入到 v4.0.0 及以上版本的 TiDB 集群。如果尝试使用 Local-backend 导入到 v2.x 或 v3.x 的集群，就会报以上错误。此时可以修改配置使用 Importer-backend 或 TiDB-backend 进行导入。
```

File: tidb-lightning/tidb-lightning-faq.md [hunk 17]
```diff
@@ -340,3 +340,18 @@ tidb-lightning-ctl --config conf/tidb-lightning.toml --checkpoint-error-destroy=
...
+
+部分 `nightly` 版本的 TiDB 集群的版本可能类似 "4.0.0-beta.2" 这样，这样的版本实际支持 Local-backend，如果使用 `nightly` 版本遇到该报错，可以通过设置配置  `check-requirements = false` 跳过版本检查。在设置此参数之前，请确保 TiDB Lightning 的配置支持对应的版本，否则无法保证导入成功。
```
Comment (TomShawn) [inline tidb-lightning/tidb-lightning-faq.md:348]:
```suggestion
部分 `nightly` 版本的 TiDB 集群的版本可能类似于 v4.0.0-beta.2。这些的版本实际支持 Local-backend。如果使用 `nightly` 版本遇到该报错，可以通过设置配置  `check-requirements = false` 跳过版本检查。在设置此参数之前，请确保 TiDB Lightning 的配置支持对应的版本，否则无法保证导入成功。
```

File: tidb-lightning/tidb-lightning-faq.md [hunk 18]
```diff
@@ -340,3 +340,18 @@ tidb-lightning-ctl --config conf/tidb-lightning.toml --checkpoint-error-destroy=
...
+
+## `restore table test.district failed: unknown columns in header [...]`
+
+出现该错误通常是因为 CSV 格式的数据文件不包含 header (第一行也是数据)，因此需要在 Lightning 的配置文件中增加如下配置项：
```
Comment (TomShawn) [inline tidb-lightning/tidb-lightning-faq.md:352]:
```suggestion
出现该错误通常是因为 CSV 格式的数据文件不包含 header（第一行也是数据），因此需要在 TiDB Lightning 的配置文件中增加如下配置项：
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**